.github/workflows/contracts.yml

```diff
@@ -33,22 +36,23 @@ jobs:
       - name: Prepare target
         run: rustup target add wasm32-unknown-unknown
 
-      # This will attempt to build all contracts in the workspace.
-      # This command will likely change when we have more than contracts in the workspace
-      - name: Build All
-        run: cargo build --release --target wasm32-unknown-unknown --verbose
+      - name: Build
+        run: cargo build -p ${{ matrix.contract }} --release --target wasm32-unknown-unknown --verbose
 
       - name: Lint
-        run: cargo clippy --all-targets --all-features
+        run: cargo clippy -p ${{ matrix.contract }} --all-targets --all-features
       - name: Upload compiled contracts
         uses: actions/upload-artifact@v3
         with:
           name: wasm-files
-          path: target/wasm32-unknown-unknown/release/*.wasm
+          path: target/wasm32-unknown-unknown/release/${{ matrix.contract }}.wasm
```

Comment:
I am not sure what you mean by affect.

Reply:
I think this artifacts are currently shared across github workflow runs. So if 2 PRs are happening in parallel, the former artifacts will be inmediatelly replaced by the second, affecting the former further jobs. For solving this, my suggestion is to add a prefix with the workflow number , so it would be something like :

`${{ github.run_number }}-wasm-files`

Reply:
Artifacts are tied to a workflow:
https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts

Reply:
Great !